COMPILING.md

```diff
@@ -1,19 +1,125 @@
+[referenced here](https://github.com/diffblue/cbmc/blob/develop/.github/workflows/pull-request-checks.yaml).
+
+The environments below have been used successfully in the
+past, but are not actively tested:
+
 - Solaris 11
 - FreeBSD 11
 - Cygwin
-- Microsoft Visual Studio
 
-The rest of this document is split up into three parts: compilation on Linux,
-MacOS, Windows.  Please read the section appropriate for your machine.
+# Building using CMAKE
+
+Building with CMake is supported across Linux, MacOS X and Windows with Visual
+Studio 2019. There are also hand-written make files which can be used to build
+separate binaries independently. Usage of the hand-written make files is
+explained in a separate section. The CMake build can build the complete
+repository in fewer steps and supports better integration with various IDEs and
+static-analysis tools. On Windows, the CMake build has the advantage of not
+depending on Cygwin or MinGW, and doesn't require manual modification of build
+files.
+
+1. Ensure you have all the build dependencies installed. Build dependencies are
+   the same as for the makefile build, but with the addition of CMake version
+   3.2 or higher. The installed CMake version can be queried with `cmake
+   --version`. To install cmake:
+   - On Debian-like distributions, do
+     ```
+     apt-get install g++ gcc flex bison make git curl patch cmake
+     ```
+   - On Red Hat/Fedora or derivates, do
+     ```
+     dnf install gcc gcc-c++ flex bison curl patch cmake
+     ```
+   - On macOS, do
+     ```
+     xcode-select --install
+     ```
+     You should also install [Homebrew](https://brew.sh), after which you can
+     run `brew install cmake` to install CMake.
+   - On platforms where installing the Java Development Kit and Maven is
+     difficult, you can avoid needing these dependencies by not building
+     JBMC. Just pass `-DWITH_JBMC=OFF` to cmake in step (4) below.
+   - On Windows, ensure you have Visual Studio 2019 or later installed. The
+     developer command line that comes with Visual Studio 2019 has cmake
+     already available. You will also need to ensure that you have winflexbison
+     installed and available in the path. winflexbison is available from
+     [the github release page](https://github.com/lexxmark/winflexbison/releases/)
+     or through
+     [the chocolatey package manager.](https://community.chocolatey.org/packages/winflexbison3)
+     Installing strawberryperl is advised if you want to run tests on Windows.
+   - Use of CMake has not been tested on Solaris or FreeBSD. However, it should
+     be possible to install CMake from the system package manager or the
+     [official download page](https://cmake.org/download) on those systems.
+     The dependencies (as listed in the relevant sections above) will also be
+     required, and should be installed using the suggested methods.
+2. Navigate to the *top level* folder of the project. This is different from
```

Comment:
Maybe add the second sentence as a bullet point, or add the paths in parentheses (`cbmc` vs `cbmc/src/`) to make it a bit less confusing?